hummingbot/strategy/dev_5_vwap/dev_5_vwap.pyx

```diff
@@ -0,0 +1,474 @@
+                    f"({market_order_record.amount} {market_order_record.base_asset}) has been filled."
+                )
+            self._has_outstanding_order = False
+
+    cdef c_did_fail_order(self, object order_failed_event):
+        if self._is_vwap:
+            self.c_check_last_order(order_failed_event)
+
+    cdef c_did_cancel_order(self, object cancelled_event):
+        if self._is_vwap:
+            self.c_check_last_order(cancelled_event)
+
+    cdef c_did_expire_order(self, object expired_event):
+        if self._is_vwap:
+            self.c_check_last_order(expired_event)
+
+    cdef c_start(self, Clock clock, double timestamp):
+        StrategyBase.c_start(self, clock, timestamp)
+        self.logger().info(f"Waiting for {self._time_delay} to place orders")
+        self._previous_timestamp = timestamp
+        self._last_timestamp = timestamp
+
+    cdef c_tick(self, double timestamp):
+        """
+        Clock tick entry point.
+
+        For this strategy, this function simply checks for the readiness and connection status of markets, and
+        then delegates the processing of each market info to c_process_market().
+
+        :param timestamp: current tick timestamp
+        """
+        StrategyBase.c_tick(self, timestamp)
+        cdef:
+            int64_t current_tick = <int64_t>(timestamp // self._status_report_interval)
+            int64_t last_tick = <int64_t>(self._last_timestamp // self._status_report_interval)
+            bint should_report_warnings = ((current_tick > last_tick) and
+                                           (self._logging_options & self.OPTION_LOG_STATUS_REPORT))
+            list active_maker_orders = self.active_maker_orders
+
+        try:
+            if not self._all_markets_ready:
+                self._all_markets_ready = all([market.ready for market in self._sb_markets])
+                if not self._all_markets_ready:
+                    # Markets not ready yet. Don't do anything.
+                    if should_report_warnings:
+                        self.logger().warning(f"Markets are not ready. No market making trades are permitted.")
+                    return
+
+            if should_report_warnings:
+                if not all([market.network_status is NetworkStatus.CONNECTED for market in self._sb_markets]):
+                    self.logger().warning(f"WARNING: Some markets are not connected or are down at the moment. Market "
+                                          f"making may be dangerous when markets or networks are unstable.")
+
+            for market_info in self._market_infos.values():
+                self.c_process_market(market_info)
+        finally:
+            self._last_timestamp = timestamp
+
+    cdef c_check_last_order(self, object order_event):
+        """
+        Check latest order to see it if was completed. If not completed, set self._has_outstanding_order to False
```

Comment:
The function is doing the right thing, but the comments here is a bit misleading... the `if` statement in this function is really just checking whether the order was made by this strategy or not - it's possible for a market connector to emit order events that were not emitted by the strategy object. e.g. the order may be made by the user manually, or made by another bot on the same account.